File: Golang/SameTree/SameTree_test.go
package SameTree

import "testing"

func TestIsSameTree(t *testing.T) {

	if isSameTree(nil, nil) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	tn1 := &TreeNode{
		Val: 1,
	}
	if isSameTree(nil, tn1) == false {
		t.Log("pass")
	} else {
		t.Error("failed")
	}
	if isSameTree(tn1, nil) == false {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	tn2 := &TreeNode{
		Val: 2,
	}
	if isSameTree(tn1, tn2) == false {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if isSameTree(&TreeNode{
		Val: 1,
		Left: &TreeNode{
			Val: 1,
		},
		Right: &TreeNode{
			Val: 1,
		},
	}, &TreeNode{
		Val: 1,
		Left: &TreeNode{
			Val: 1,
		},
		Right: &TreeNode{
			Val: 1,
		},
	}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/SameTreeTest.py
from unittest import TestCase

from SameTree import SameTree, TreeNode


class TestSameTree(TestCase):
    def test_isSameTree(self):
        st = SameTree()

        self.assertTrue(st.isSameTree(None, None))

        tn1 = TreeNode(1)
        self.assertFalse(st.isSameTree(None, tn1))
        self.assertFalse(st.isSameTree(tn1, None))

        tn2 = TreeNode(2)
        self.assertFalse(st.isSameTree(tn1, tn2))

        tn_1_1_1_1 = TreeNode(1)
        tn_1_1_1_1.left = TreeNode(1)
        tn_1_1_1_1.right = TreeNode(1)
        tn_2_1_1_1 = TreeNode(1)
        tn_2_1_1_1.left = TreeNode(1)
        tn_2_1_1_1.right = TreeNode(1)
        self.assertTrue(st.isSameTree(tn_1_1_1_1, tn_2_1_1_1))
